multicluster/controllers/multicluster/leader/stale_controller.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2023 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package leader
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"antrea.io/antrea/multicluster/apis/multicluster/constants"
+	mcv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+	"antrea.io/antrea/multicluster/controllers/multicluster/common"
+	"antrea.io/antrea/multicluster/controllers/multicluster/commonarea"
+)
+
+const (
+	indexKey                       = "spec.clusterID"
+	memberClusterAnnounceStaleTime = 2 * time.Hour
+)
+
+var getResourceExportsByClusterIDFunc = getResourceExportsByClusterID
+
+// LeaderStaleResCleanupController will run periodically (memberClusterAnnounceStaleTime / 2 ) (1 Hour)
+// to clean up stale MemberClusterAnnounce resources in the leader cluster if the MemberClusterAnnounce
+// timestamp annotation has not been updated for memberClusterAnnounceStaleTime (2 Hours).
```

Comment:
Yeah, I was thinking it may help to cleanup stale resource quicker with 2 hours, but it may not help a lot, change it back to 24 hours. We may find a new solution to guarantee MemberClusterAnnounce can be deleted from a member cluster in any cases, then this can be removed in the future.